airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/MongoUtil.java

```diff
@@ -163,6 +163,7 @@ public static Optional<CollectionStatistics> getCollectionStatistics(final Mongo
       try (final MongoCursor<Document> cursor = output.allowDiskUse(true).cursor()) {
         if (cursor.hasNext()) {
           final Document stats = cursor.next();
+          @SuppressWarnings("unchecked")
```

Comment:
nasty one here. stats.get returns an Object by default. There is a get(String, Class<T>) that returns a T, but that doesn't work with a parameterized class, so no matter what, there's a suppresswarnings here :(